src/frontend/src/flows/authenticate/index.ts

```diff
@@ -0,0 +1,355 @@
+        <button type="button" class="linkStyle" id="manageButton">
+          Manage your Identity Anchor
+        </button>
+      </div>
+      <hr />
+    </div>
+    ${navbar}
+  </div>
+  ${footer}`;
+
+const existingAnchorSection = (userNumber: bigint) => html` <div
+  class="modeContainer"
+>
+  <div class="childContainer">
+    <div id="identityAnchor" class="highlightBox anchorText">${userNumber}</div>
+    <button id="editAnchorButton">${editIcon}</button>
+  </div>
+</div>`;
+
+const editAnchorSection = (userNumber?: bigint) => html` <div
+  class="modeContainer"
+>
+  <div class="childContainer">
+    <input
+      class="anchorText"
+      type="text"
+      id="userNumberInput"
+      placeholder="Enter anchor"
+      value="${userNumber !== undefined ? userNumber?.toString() : ""}"
+    />
+  </div>
+  <div id="invalidAnchorMessage" class="error-message-hidden smallText">
+    The Identity Anchor is not valid. Please try again.
+  </div>
+</div>`;
+
+export interface AuthSuccess {
+  userNumber: bigint;
+  connection: IIConnection;
+  sendDelegationMessage: () => void;
+}
+
+/**
+ * Shows the authorize application view and returns information about the authenticated user and a callback, which sends
+ * the delegation to the application window. After having received the delegation the application will close the
+ * Internet Identity window.
+ */
+export default async (): Promise<AuthSuccess> => {
+  return new Promise((resolve) => {
+    withLoader(async () => {
+      const authContext = await waitForAuthRequest();
+      if (authContext === null) {
+        // The user has manually navigated to "/#authorize".
+        redirectToWelcomeScreen();
+        return;
+      }
+      const userNumber = getUserNumber();
+      init(authContext, userNumber === undefined, userNumber).then(resolve);
+    });
+  });
+};
+
+const init = (
+  authContext: AuthContext,
+  editMode: boolean,
+  userNumber?: bigint
+): Promise<AuthSuccess> => {
+  displayPage(authContext.requestOrigin, editMode, userNumber);
+  const manageButton = document.getElementById(
+    "manageButton"
+  ) as HTMLButtonElement;
+  manageButton.onclick = () => redirectToWelcomeScreen();
+  initRecovery(editMode);
+
+  return new Promise((resolve) => {
+    const authenticateButton = document.getElementById(
+      "authorizeButton"
+    ) as HTMLButtonElement;
+    if (editMode) {
+      const userNumberInput = document.getElementById(
+        "userNumberInput"
+      ) as HTMLInputElement;
+      userNumberInput.select();
+      userNumberInput.onkeypress = (e) => {
+        if (e.key === "Enter") {
+          // authenticate if user hits enter
+          e.preventDefault();
+          authenticateButton.click();
+        }
+      };
+    } else {
+      const editAnchorButton = document.getElementById(
+        "editAnchorButton"
+      ) as HTMLButtonElement;
+      editAnchorButton.onclick = () =>
+        init(authContext, true, userNumber).then(resolve);
```

Comment:
(paranoia) here there's an assumption that the `userNumber` never changes; but I can see us in the future adding a "tickmark" button that disables editMode, in which case if the user (1) lands on the page (2) edits the anchor (3) clicks the tickmark (4) re-tries to edit the anchor, the anchor will be reverted to the value from (1).

Instead of having state duplicated (once in the input, and once in js) we could use the value in the `input` as source of truth?